### PR TITLE
Add model hub catalog v2 for model hub

### PR DESCRIPTION
**Pull request by smothiki**

### CODE REVIEW

File: model-hub-catalog-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,681 @@
+# The model spec is designed keeping in mind both the HF and NGC APIs and tried to capture minimal information from both the APIs.
+models:
+  - name: llama3 instruct 
+    displayName: Llama 3 
+    description: Llama-3 instruct model published by Meta
+    modelVariants:
+      - variantId: llama 3 8b instruct
+        displayName: 8b
+        optimizationProfiles:
+        - profileId: ngc://nim/meta/llama3-8b-instruct:0.10.0+cbc614f5-l40sx1-fp8-throughput
```
Comment (smothiki) [inline model-hub-catalog-v2.yaml]: Remove ngc://

File: model-hub-catalog-v2.yaml [hunk 2]
```diff
@@ -0,0 +1,681 @@
...
+          displayName: llama 3 8b instruct l40sx1 fp8 throughput
+          framework: TensorRT-LLM
+          sha: 09e2f8e68f78ce94bf79d15b40a21333cea5d09dbe01ede63f6c957f4fcfab7b
+          modelFormat: trl-llm
```
Comment (smothiki) [inline model-hub-catalog-v2.yaml:18]: add ngcMetadata key and value should be ngc catalog base64 encoded entry

File: model-hub-catalog-v2.yaml [hunk 3]
```diff
@@ -0,0 +1,681 @@
...
+        requireToken: True
+        displayName: 7b
+        createdDate: "2024-0522T9:56:38"
+        updatedDate: "2024-05-22T16:34:28"
+        optimizationProfiles:
+        - profileId: mistralai/Mistral-7B-v0.3
+          displayName: Mistral 7B v0.3
+          sha: b67d6a03ca097c5122fa65904fce0413500bf8c8
+          framework: transformers
+          modelFormat: N/A
```
Comment (smothiki) [inline model-hub-catalog-v2.yaml:656]: requireLicense is also needed

File: model-hub-catalog-v2.yaml [hunk 4]
```diff
@@ -0,0 +1,681 @@
...
+      - "Language Generation"
+      - "NeMo"
+      - "NVIDIA Validated"
+    config:
+      architectures:
+        - Other
+      modelType: llama
+    license: NVIDIA AI Foundation Models Community License
+    source:
+      URL: https://catalog.ngc.nvidia.com/orgs/nvidia/teams/ai-foundation/collections/llama2
```
Comment (smothiki) [inline model-hub-catalog-v2.yaml]: change the URL to llama3

Comment (smothiki) [inline model-hub-catalog-v2.yaml]: Move source to the variant level

**Review by dchung-work — APPROVED** [review]

File: model-hub-catalog-v2.yaml [hunk 5]
```diff
@@ -0,0 +1,718 @@
...
+models:
+  - name: llama3 instruct 
+    displayName: Llama 3 
+    description: Llama-3 instruct model published by Meta
+    modelVariants:
+      - variantId: llama 3 8b instruct
+        displayName: 8b
+        source:
+          URL: https://catalog.ngc.nvidia.com/orgs/nim/teams/meta/models/llama3-8b-instruct
+          type: NGC
```
Comment (dchung-work) [inline model-hub-catalog-v2.yaml]: Can we move `type` to the top level? There's no need to have `type` for each `modelVarant`

**Review by dchung-work — APPROVED** [review]

**Review by anirudh9373 — APPROVED** [review]

**Pull request merged**